src/_argon2_cffi_bindings/_ffi_build.py

```diff
@@ -27,7 +29,9 @@ def _get_target_platform(arch_flags, default):
 
 
 target_platform = _get_target_platform(
-    os.environ.get("ARCHFLAGS", ""), platform.machine()
+    os.environ.get("ARCHFLAGS", ""),
+    platform.machine(),
+    os.environ.get("VSCMD_ARG_TGT_ARCH", ""),
```

Comment:
This was one way to ensure the build correctly gets the target platform when cross-compiling at least when using cibuildwheel since cibuildwheel sets `VSCMD_ARG_TGT_ARCH` when it runs a cross-compiled build on Windows. However, another possible way could be to `import setuptools` and then use `get_platform` instead of `platform.machine()` but then that introduces a setuptools dependency for this build script. Any thoughts on what might be better?

Reply:
I'm not enthusiastic about adding dependencies, but this is the build step, correct? IOW: it already depends on setuptools?

That said, it looks like the API is actually `pkg_resources.get_build_platform()` and that module has been wholesale deprecated and I'd rather avoid adding deprecated API usage.

---

Sorry for the late reply, I've been traveling and only catching up. I've done some rejiggering to allow for contributions so don't forget to pull.

But it's looking like a blessing in hindsight, since GHA added ARM64 Windows support three days ago: https://blogs.windows.com/windowsdeveloper/2025/04/14/github-actions-now-supports-windows-on-arm-runners-for-all-public-repos/

So it should be possible to clean this all up?

Reply:
No worries! I took a little while to see that you had responded as well! Yeah, I'll get go ahead and update this to use the arm64 runner instead and clean this up.